src/query/expression/src/types.rs

```diff
@@ -176,3 +179,19 @@ pub trait ArgType: ValueType {
         Self::build_column(col)
     }
 }
+
+#[macro_export]
+macro_rules! with_basic_no_number_mapped_type {
+    (| $t:tt | $($tail:tt)*) => {
+        match_template::match_template! {
+            $t = [
+                String => StringType,
+                Boolean => BooleanType,
+                // TODO(RinChanNOW): enable TimestampType
+                // Need to fix: TimestampType cannot satisfy the bounded trait of `AggregateMinMaxAnyFunction`
+                // Timestamp => TimestampType,
+            ],
+            $($tail)*
+        }
+    }
+}
```

Comment:
Need some help: TimestampType cannot satisfy the bounded trait of AggregateMinMaxAnyFunction. Don't know why.

Reply:
ScalarRef of `TimestampType ` did satisfy?

Reply:
> ScalarRef of TimestampType did not satisfy?

Probably.

Haven't figured out what is wrong yet.

The compiler said:

```
= note: the following trait bounds were not satisfied:
      `ScalarState<TimestampType, CmpAny>: ScalarStateFunc<TimestampType>`
```

(CmpMax, CmpMin are the same)

Reply:
Solved. `Timestamp` didn't impl `Default`...

Reply:
Or we can remove the `T::Scalar: Default` bound.